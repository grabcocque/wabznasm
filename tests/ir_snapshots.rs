//! Snapshot tests for LLVM IR codegen using insta
//! NOTE: codegen module not available, tests disabled

/*
use insta::assert_snapshot;
use wabznasm::codegen::dump_ir;

/// Simple addition IR snapshot (template-based)
#[test]
fn ir_snapshot_addition() {
    let ir = dump_ir("1+2").unwrap();
    assert_snapshot!(ir);
}

/// Factorial IR snapshot (template-based)
#[test]
fn ir_snapshot_factorial() {
    let ir = dump_ir("5!").unwrap();
    assert_snapshot!(ir);
}

/// Operator precedence IR snapshot (template-based)
#[test]
fn ir_snapshot_precedence_mul_add() {
    let ir = dump_ir("2*3+4").unwrap();
    assert_snapshot!(ir);
}
*/
