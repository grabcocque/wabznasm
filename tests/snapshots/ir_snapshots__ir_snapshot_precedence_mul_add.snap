---
source: tests/ir_snapshots.rs
expression: ir
---
; ModuleID = 'calc_jit'
source_filename = "calc_jit"

define i64 @main() {
entry:
  ret i64 10
}
